### PR TITLE
Merge ParserContextFlags into NodeFlags

### LINKED ISSUE #6578: Remove `ParserContextFlags` and unify with `NodeFlags`
`parserContextFlags` is odd, but we're "out of room" in `NodeFlags` so they can't be directly merged.

Figure out a strategy to redivide these into a sensible set of flagsets based on how they're used in our codebase.

### PR DESCRIPTION
**Pull request by ahejlsberg**
Fixes #6578.

### CODE REVIEW

File: src/compiler/types.ts [hunk 1]
```diff
@@ -979,6 +946,7 @@ namespace ts {
     // @kind(SyntaxKind.ArrayLiteralExpression)
     export interface ArrayLiteralExpression extends PrimaryExpression {
         elements: NodeArray<Expression>;
+        multiLine?: boolean;
```
Comment (DanielRosenwasser) [inline src/compiler/types.ts]: Should these be `/*@internal*/`?

File: src/compiler/types.ts [hunk 2]
```diff
@@ -938,6 +904,7 @@ namespace ts {
         text: string;
         isUnterminated?: boolean;
         hasExtendedUnicodeEscape?: boolean;
+        isOctalLiteral?: boolean;
```
Comment (DanielRosenwasser) [inline src/compiler/types.ts]: Actually, I think these should be `/*@internal*/`

Comment (ahejlsberg) [inline src/compiler/types.ts]: Agreed.

**Comment by RyanCavanaugh** [comment]
:+1:

**Comment by sandersn** [comment]
:+1:

**Pull request merged**